src/common/transformations/tests/type_relaxed_tests.cpp

```diff
@@ -411,17 +412,38 @@ bool fuse_type_to_convert_cpu(const std::shared_ptr<ov::Node>& node, const preci
     const auto& to = it->second;
     if (auto convert = ov::as_type_ptr<ov::opset1::Convert>(node)) {
         // For Convert node, converting precision from floating point to boolean will lead to mathematical
-        // error, because here the output precision boolean is replaced by u8. E.g. floating point value 0.01
-        // is converted to be 1 for boolean, but 0 for u8. Thus an Abs and Ceil node should be added before the
-        // Convert node for this scenario.
+        // error, because here the output precision boolean is replaced by u8:
+        //  - floating point value 0.01 is converted to be 1 for boolean, but 0 for u8 - need to insert Ceil.
+        //  - floating point value 256 is converted to be 1 for boolean, but 0 for u8 - need to insert Min(x, UINT8_MAX)
+        //  - floating point value -256 is converted to be 1 for boolean, but 0 for u8 - need to insert Abs before Min.
+        // Thus an Abs, Ceil and Min nodes should be added before the Convert node for this scenario.
         if (convert->input(0).get_element_type().is_real() &&
             convert->get_convert_element_type() == ov::element::boolean && to.is_integral_number()) {
-            auto abs = std::make_shared<ov::opset1::Abs>(convert->input_value(0).get_node_shared_ptr());
-            auto ceil = std::make_shared<ov::opset1::Ceiling>(abs);
-            auto new_convert = std::make_shared<ov::opset1::Convert>(ceil, to);
+            ov::NodeVector new_ops;
+#define CREATE_OP(NAME, TYPE, ...)                          \
+    const auto& NAME = std::make_shared<TYPE>(__VA_ARGS__); \
+    new_ops.push_back(NAME);
+
+            const auto& in_prec = convert->get_input_element_type(0);
+            auto data = convert->input_value(0).get_node_shared_ptr();
+            auto item = precisions.find(in_prec);
+            if (item != precisions.end()) {
+                // Add convert node for unsupported precision, such as FP64
+                CREATE_OP(pre_convert, ov::opset1::Convert, data, item->second);
+                data = pre_convert;
+            }
+
+            CREATE_OP(abs, ov::opset1::Abs, data);
```

Comment:
I have some doubts about using the "CREATE_OP" macro.
I think it makes the code less readable, hard to understand connections between nodes.

I would recommend to use ov::pass::NodeRegistry class, e.g.
``` 
NodeRegistry reg;
auto ov_const = reg.make<ov::opset1::Constant>(...args); // example of shared_ptr<Constant> creation
...
ov::copy_runtime_info(convert, reg.get());
```

Reply:
Cool suggestion! I didn't know about `NodeRegistry`!
Applied, thanks!